src/opm/parser/eclipse/EclipseState/Schedule/Well/Well.cpp

```diff
@@ -776,12 +775,22 @@ bool Well::canOpen() const {
         if (prod.WaterRate.is<std::string>())
           return true;
 
-        return ((prod.OilRate.get<double>() + prod.GasRate.get<double>() + prod.WaterRate.get<double>()) != 0);
+        if (!prod.OilRate.zero())
```

Comment:
I am not totally sure about the usage and associated implementation here, so I might understand wrong. 

For the rate constraints, the most restrict one will work as the effective one. For example, if we have input constraints for all the three phases, one of them is zero, then this well will work with the zero constraint, which means, when we do not allow crossflow, it should return `false` here. 

The current logic is that the well can be opened if we have one non-zero rate constraint. 

I admit it is kind of corner case we do not encounter frequently.  And the difficulty comes from how we implement the default constraints, which in theory is not constraint.

Reply:
You are probably right, but this PR does not imply any change (since we can not have negative rates).